fenics_ice/model.py

```diff
@@ -148,7 +148,17 @@ def alpha_from_data(self):
     def bglen_from_data(self):
         """Get bglen field from initial input data"""
         self.bglen = self.input_data.interpolate("Bglen", self.Q)
-
+        """Get bglen mask field from input data"""
+        self.bglen_mask = self.input_data.interpolate("Bglenmask", self.M)
```

Comment:
Can the `interpolate` method be used safely with values that may be NaN? If not we should first convert to zero/non-zero before interpolation.

Reply:
as long as I am aware these two functions [interpolate()](https://github.com/EdiGlacUQ/fenics_ice/blob/3c63d1fe968aeebbce8d0ac10a1af578b768f4df/fenics_ice/model.py#L232-L257) and the one starting on L232 do not handle nan's as stated in the comments.
```
def interpolate(values, vtx, wts):
            """Bilinear interpolation, given vertices & weights above"""
            return np.einsum('nj,nj->n', np.take(values, vtx), wts)
```

Reply:
I wanted to ensure that any cell that could possibly be affected would have its DoF as Nan. @bearecinos i believe this interpolate function is defined in inout.py

Reply:
@jrmaddison  see new changes to this function